addons/website_google_map/controllers/main.py

```diff
@@ -24,14 +24,23 @@ class GoogleMap(http.Controller):
 
     @http.route(['/google_map'], type='http', auth="public", website=True, sitemap=False)
     def google_map(self, *arg, **post):
+        PartnerSudo = request.env['res.partner'].sudo()
         clean_ids = []
-        for partner_id in post.get('partner_ids', "").split(","):
-            try:
-                clean_ids.append(int(partner_id))
-            except ValueError:
-                pass
-        partners = request.env['res.partner'].sudo().search([("id", "in", clean_ids),
-                                                             ('website_published', '=', True), ('is_company', '=', True)])
+        domain = [('website_published', '=', True)]
+        if post.get('partner_ids'):
+            for partner_id in post['partner_ids'].split(","):
+                try:
+                    clean_ids.append(int(partner_id))
+                except ValueError:
+                    pass
+            domain += [("id", "in", clean_ids), ('is_company', '=', True)]
+        elif post.get('fn') and hasattr(self, f"_google_map_domain_{post['fn']}"):
+            domain += getattr(self, f"_google_map_domain_{post['fn']}")(**post)
```

Comment:
This is triggering the rd-security review stuff, I'll let you decide. Maybe another solution would be to extract the domain computation part in a sub-method, then instead of part of a function name, it would be a boolean like `use_XXX_domain`. And in your specific app override, instead of creating `_google_map_domain_XXX`, you would extend the sub-method that creates the domain with `if use_XXX_domain: ...`. That would be as generic as your solution.